Change how we retrieve rule descriptions when generating the inclusive language database

agnieszkaszuba opened the pull request:
## Context
<!--
What do we want to achieve with this PR? Why did we write this code?
-->

* Previously, we retrieved the inclusive language assessment rule descriptions by using a regex to extract them from the functions used to apply the rule. This method was unreliable and didn't work for all rules. Instead, in this PR, a new property is added to each inclusive language assessment with a rule that contains the rule description.

## Summary

<!--
Attach one of the following labels to the PR: `changelog: bugfix`, `changelog: enhancement`, `changelog: other`, `changelog: non-user-facing`.
If the changelog item is a bugfix, please use the following sentence structure: Fixes a bug where ... would ... (when ...).
If the changelog item is meant for the changelog of another add-on, start your changelog item with the name of that add-on's repo between square brackets, for example: * [wordpress-seo-premium] Fixes a bug where ....
If the changelog items is meant for the changelog of a javascript package, specify between square brackets in which package changelog the item should be included, for example: * [@yoast/components] Fixes a bug where ....
If the same changelog item is applicable to multiple changelogs/add-ons, add a separate changelog item for all of them.
-->
This PR can be summarized in the following changelog entry:

* Adds a rule description property to each inclusive language assessment that has a rule.
* [shopify-seo] Adds a rule description property to each inclusive language assessment that has a rule.

## Relevant technical choices:

* I chose to duplicate the list of precedence/following exceptions that are passed to the functions that apply the rules, and to the function that generates the rule description. Rather than extracting them to a separate variable. I thought that the benefit of avoiding duplication doesn't outweigh the benefit of being able to see the exception lists next to the other assessment data, rather than somewhere else. This way, we can find the information quicker, as well as more easily spot any mistakes or missing exceptions. 

## Test instructions
<!--
Please follow these guidelines when creating test instructions:
- Please provide step-by-step instructions how to reproduce the issue, if applicable.
- Write step-by-step test instructions aimed at non-tech-savvy users, even if the PR is not user-facing.
-->
### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Generate the inclusive language database following these instructions: [How to generate the inclusive language database](https://yoast.atlassian.net/wiki/spaces/LIN/pages/2477621286/Inclusive+language+database).
* Confirm that the rules for the assessments are outputted correctly. They should be the same as in this file: [Inclusive language database test](https://docs.google.com/spreadsheets/d/1uTgmQNuzHd2EnkZunjv0Nhe9qbsHSXe6nkEr0fDH4dA/edit?usp=sharing).
* Smoke test the inclusive language analysis
     * Create a post and add some phrases that have rules, for example: `seniors`, `I'm so OCD`, `alcoholics`, `normal person`.
     * Confirm that the inclusive language analysis returns feedback for all those phrases.
     * Change some of the phrases so that they are considered inclusive according to the rules, for example: `high school seniors`, `I have OCD`, `alcoholics anonymous`.
     * Confirm that no feedback shows for these phrases.

#### Relevant test scenarios
* [x] Changes should be tested with the browser console open
* [ ] Changes should be tested on different posts/pages/taxonomies/custom post types/custom taxonomies
* [ ] Changes should be tested on different editors (Default Block/Gutenberg/Classic/Elementor/other)
* [ ] Changes should be tested on different browsers
* [ ] Changes should be tested on multisite
<!--
If you have checked any of the above cases, please add some context about the reason, what to check in the console,
which type/editor/browser should be tested in particular, multisite with subfolders or subdomains, etc.
-->

### Test instructions for QA when the code is in the RC
<!--
Sometimes some steps from the test instructions for the acceptance test aren't relevant anymore once the code has been merged or the feature is complete. If that is the case, do not check the checkbox below.
QA is our Quality Assurance team. The RC is the release candidate zip that is tested before a release
-->

* [ ] QA should use the same steps as above.

<!--
If the above checkbox has not been checked, write down all steps QA should take to test this PR, not only the difference with the acceptance test steps. If QA should use the test instructions specified on the epic, paste a link to the relevant comment on the epic.
-->
QA can test this PR by following these steps:

* Use the same steps as above, except for the first two steps which do not need to be tested by QA.

## Impact check
<!--
Sometimes PRs have a bigger impact than is suggested in the user-facing changes. In such cases,
additional (regression) testing might be necessary. To make it clear what parts might need additional testing, please outline which parts of the plugin have been impacted by this PR.
-->
This PR affects the following parts of the plugin, which may require extra testing:

* No further impact.

## UI changes

* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Other environments

* [x] This PR also affects Shopify. I have added a changelog entry starting with `[shopify-seo]`, added test instructions for Shopify and attached the `Shopify` label to this PR.

## Documentation

* [x] I have written documentation for this change. For example, comments in the Relevant technical choices, comments in the code, documentation on Confluence / shared Google Drive / [Yoast developer portal](https://developer.yoast.com/), or other.

## Quality assurance

* [x] I have tested this code to the best of my abilities.
* [x] During testing, I had activated [all plugins that Yoast SEO provides integrations for](https://github.com/Yoast/wordpress-seo/blob/trunk/readme.txt#L106).
* [x] I have added unit tests to verify the code works as intended.
* [ ] If any part of the code is behind a feature flag, my test instructions also cover cases where the feature flag is switched off.
* [x] I have written this PR in accordance with my team's definition of done.
* [x] I have checked that the base branch is correctly set.

## Innovation

* [x] No innovation project is applicable for this PR.
* [ ] This PR falls under an innovation project. I have attached the `innovation` label.
* [ ] I have added my hours to [the WBSO document](http://yoa.st/wbso).

Fixes [Improve script for generating inclusive language database so that all rules are outputted](https://github.com/Yoast/lingo-other-tasks/issues/155)

Jordi-PV commented on packages/yoastseo/spec/scoring/assessments/inclusiveLanguage/exportInclusiveLanguageConfigSpec.js:
```diff
@@ -89,27 +79,12 @@ describe( "Export of the inclusive language configuration", () => {
 
 		// Set doExport to true to write the results to a temporary file.
-		const doExport = false;
+		const doExport = true;
```
Shouldn't this value be set back to false before merge?

agnieszkaszuba replied:
Yes, thanks!

Jordi-PV commented:
AT 👍 It works like a charm 
CR 👍 (i just left one small comment before merging it)

Jordi-PV commented:
AT 👍 It works like a charm 
CR 👍 (i just left one small comment before merging it)

Jordi-PV commented on packages/yoastseo/src/scoring/assessments/inclusiveLanguage/InclusiveLanguageAssessment.js:
```diff
@@ -36,7 +36,7 @@ export default class InclusiveLanguageAssessment {
 	 * @returns {void}
```
Also good to add the description of these parameters:  

	 * @param {string} [config.ruleDescription] A description of the rule.
	 * @param {string} [config.category] The category of the assessment.

agnieszkaszuba commented:
Thanks, I've implemented both your suggestions! About the error, I think I've always had it when I run the command, so it is not a problem. I added to the documentation that this error is expected.
